Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp

```diff
@@ -72,45 +73,16 @@ namespace AZ::Internal
             struct EngineInfo
             {
                 AZ::IO::FixedMaxPath m_path;
-                AZ::SettingsRegistryInterface::FixedValueString m_moniker;
+                FixedValueString m_moniker;
             };
 
             struct EnginePathsVisitor : public AZ::SettingsRegistryInterface::Visitor
             {
                 void Visit(
-                    [[maybe_unused]] AZStd::string_view path, [[maybe_unused]] AZStd::string_view valueName,
-                    [[maybe_unused]] AZ::SettingsRegistryInterface::Type type, AZStd::string_view value) override
-                {
-                    m_enginePaths.emplace_back(EngineInfo{AZ::IO::FixedMaxPath{value}.LexicallyNormal(), {}});
-                }
-
-                AZ::SettingsRegistryInterface::VisitResponse Traverse(
                     [[maybe_unused]] AZStd::string_view path, AZStd::string_view valueName,
-                    AZ::SettingsRegistryInterface::VisitAction action, AZ::SettingsRegistryInterface::Type type) override
+                    [[maybe_unused]] AZ::SettingsRegistryInterface::Type type, AZStd::string_view value) override
                 {
-                    auto response = AZ::SettingsRegistryInterface::VisitResponse::Continue;
-                    if (action == AZ::SettingsRegistryInterface::VisitAction::Begin)
-                    {
-                        if (type == AZ::SettingsRegistryInterface::Type::Array)
-                        {
-                            if (valueName.compare("engines") != 0)
-                            {
-                                response = AZ::SettingsRegistryInterface::VisitResponse::Skip;
-                            }
-                        }
-                    }
-                    else if (action == AZ::SettingsRegistryInterface::VisitAction::Value)
-                    {
-                        if (type == AZ::SettingsRegistryInterface::Type::String)
-                        {
-                            if (valueName.compare("path") != 0)
-                            {
-                                response = AZ::SettingsRegistryInterface::VisitResponse::Skip;
-                            }
-                        }
-                    }
-
-                    return response;
+                    m_enginePaths.emplace_back(EngineInfo{ AZ::IO::FixedMaxPath{value}.LexicallyNormal(), FixedValueString{valueName} });
```

Comment:
Were these changes tested against a manifest with multiple registered engines?

Reply:
Yes, I tested this against a manifest with multiple engines and multiple formats for registration
i.e
```JSON
"engines_path":
{
   "o3de": "D:/o3de/o3de",
   "o3de-install": "D/o3de/install"
}
```